core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -722,10 +728,7 @@ private[spark] class TaskSetManager(
       put(info.executorId, clock.getTimeMillis())
     sched.dagScheduler.taskEnded(tasks(index), reason, null, null, info, taskMetrics)
     addPendingTask(index)
-    if (!isZombie && state != TaskState.KILLED && !reason.isInstanceOf[TaskCommitDenied]) {
-      // If a task failed because its attempt to commit was denied, do not count this failure
-      // towards failing the stage. This is intended to prevent spurious stage failures in cases
-      // where many speculative tasks are launched and denied to commit.
```

Comment:
as mentioned above, this comment is still relevant but now in the wrong place. We should move it `TaskCommitDenied`